dist/tools/openocd/openocd.sh

```diff
@@ -178,6 +178,13 @@ do_flash() {
 do_debug() {
     test_config
     test_elffile
+
+    # Configure halt command: prepend a reset if required by the board config
+    PRE_RESET_HALT=''
+    if [ -n "${PRE_DEBUG_RESET_HALT}" ]; then
+        PRE_RESET_HALT='reset'
+    fi
+
```

Comment:
I think instead of doing this you can evaluate the variable directly on the code, since it's used only there. Each command anyways gets executed separately.

Reply:
I prefer keeping it this way. The command prefix is defined before, so there's no inline bash code added when the openocd command is built.

Reply:
But in this case it has to be defined globally and that's exactly what I wanted to avoid.

Reply:
By the way, I changed the previous code by using a local variable. Still works.

Reply:
> But in this case it has to be defined globally and that's exactly what I wanted to avoid.

All the variables that are used in this script are global, they are available because they're exported since this script is out of the scope of the Makefiles. Thus, there's no real concept of "global" variable here, all of them are.